[CS2113T-T09-3] StockPal

NgYaoDong opened the pull request:
StockPal is a user-friendly, cost-effective solution for small business owners, providing real-time insights and eliminating manual tracking. However, it does not cater to large-scale enterprise inventory management needs or integrate with accounting systems.

wjunjie01 committed:
Update Java Doc description  for the Description class

wjunjie01 committed:
Update DG for list command feature

leongxingyu committed:
Refactor code for InflowCommand and history

leongxingyu committed:
Refactor code for outflow and history

wjunjie01 committed:
Fix list typo on DevGuide

leongxingyu committed:
Add UML diagram for inflow and outflow class

leongxingyu committed:
Add outflow command sequence diagram

leongxingyu committed:
Add inflow command sequence diagram

leongxingyu committed:
Add DG draft 1

leongxingyu committed:
Edit DG v2.0

leongxingyu committed:
Update DG

leongxingyu committed:
Refactor Inflow and Outflow code

leongxingyu committed:
Update sequence diagram for inflow and outflow class

leongxingyu committed:
Update code attribution

wjunjie01 committed:
Change icon of list class diagram

wjunjie01 committed:
Update list UML diagram to fix bugs

wjunjie01 committed:
Fix return arrow

leongxingyu committed:
Edit sequence diagram

leongxingyu committed:
Refactor code for notifying low quantities

Kobot7 committed:
Update DG with details for EditCommand

Kobot7 committed:
Update EditCommand sequence diagram

Kobot7 committed:
Update Data component and Glossary

leongxingyu committed:
Update productList code to fix bugs

leongxingyu committed:
Add JUnit testing for low quantity products

leongxingyu committed:
Fix gradle check

EdmundTangg committed:
Merge pull request #73 from wjunjie01/list

Add list UML diagram and update Developer Guide

leongxingyu committed:
Merge branch 'master' into Edit-Inflow-Outflow-and-add-DG

Kobot7 committed:
Merge pull request #76 from leongxingyu/Edit-Inflow-Outflow-and-add-DG

Edit inflow outflow and add dg

Kobot7 committed:
Merge branch 'master' into developer-guide

Kobot7 committed:
Fixed merging issue of misplaced lines

Kobot7 committed:
Delete text-ui-test

Kobot7 committed:
Remove IO redirection test from gradle.yml

cheeseong2001 committed:
Merge pull request #77 from Kobot7/developer-guide

Update DG with EditCommand and DataComponent

cheeseong2001 committed:
Update exception message

cheeseong2001 committed:
Update DeleteCommand in DeveloperGuide

Kobot7 committed:
Add individual help page feature

cheeseong2001 committed:
Implement PidNotFoundException within PID search

NgYaoDong committed:
Merge pull request #78 from cheeseong2001/branch-Delete

Update Delete feature in DeveloperGuide

cheeseong2001 committed:
Refactor parser validation methods and update method names

cheeseong2001 committed:
Update Parser component in DeveloperGuide

Kobot7 committed:
Merge pull request #79 from cheeseong2001/branch-Parser

Update Parser class and update DeveloperGuide for Parser

Kobot7 committed:
Merge branch 'master' into help-command

Kobot7 committed:
Fix merging issue

Kobot7 committed:
Fix error with parsing edit command

Kobot7 committed:
Update DG

EdmundTangg committed:
Merge pull request #80 from Kobot7/help-command

Add individual help page feature

EdmundTangg committed:
small change

EdmundTangg committed:
resolved merge conflict

EdmundTangg committed:
Updated find command to search substrings, fix output of history command, do testing for find command, changed DG

EdmundTangg committed:
edited FindCommandTest

NgYaoDong committed:
Merge pull request #83 from EdmundTangg/master

Added testings and UG change

NgYaoDong committed:
Add TransactionStorage and theme for User Guide.

NgYaoDong committed:
Reorder methods to comply with checkstyle

Kobot7 committed:
Merge pull request #84 from NgYaoDong/NgYaoDong-TransactionStorage

Add TransactionStorage and theme for User Guide.

NgYaoDong committed:
Edit theme for Github Pages

NgYaoDong committed:
Edit User Guide to show html section.

NgYaoDong committed:
Edit to show Alert box for warnings.

NgYaoDong committed:
Edit bug in alert box html

NgYaoDong committed:
Edit theme for github pages.

NgYaoDong committed:
Add minor edits to the UserGuide.

NgYaoDong committed:
Add test code to check compliance with html

NgYaoDong committed:
Add edit to Caution box for User Guide.

NgYaoDong committed:
Add emoji for Notes box

NgYaoDong committed:
Edit bug in emoji

NgYaoDong committed:
Update User Guide with table

NgYaoDong committed:
Update styling of table

NgYaoDong committed:
Update proper formatting of unordered list

NgYaoDong committed:
Update formatting of bolding of words.

cheeseong2001 committed:
Update Parser logic

cheeseong2001 committed:
Include more Parser tests

cheeseong2001 committed:
Update user guide for Delete feature

cheeseong2001 committed:
Merge pull request #85 from NgYaoDong/master

Format User Guide to show segments properly

cheeseong2001 committed:
Merge branch 'master' into branch-Parser

cheeseong2001 committed:
Anchoring link to delete section

cheeseong2001 committed:
Update accordingly to checkstyle

NgYaoDong committed:
Merge pull request #86 from cheeseong2001/branch-Parser

Update parser test cases and delete feature in user guide

NgYaoDong committed:
Add bug fixes to JSON file loading

EdmundTangg committed:
Merge pull request #87 from NgYaoDong/NgYaoDong-TransactionStorage

Add bug fixes to JSON file loading

EdmundTangg committed:
Checked PID in history, added testing for history

EdmundTangg committed:
added logging details

EdmundTangg committed:
deleted test file

leongxingyu committed:
Add UG for inflow/outflow commands

leongxingyu committed:
Merge pull request #88 from EdmundTangg/history

history command

leongxingyu committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-T09-3/tp into Add-UG

leongxingyu committed:
Edit typos and errors in UG

leongxingyu committed:
Add hyperlinks to link to Edit Command

Kobot7 committed:
Merge pull request #89 from leongxingyu/Add-UG

Add ug

Kobot7 committed:
Update help feature user guide

EdmundTangg committed:
Merge pull request #90 from Kobot7/user-guide

Update help feature user guide

wjunjie01 committed:
Add UG for list

wjunjie01 committed:
Fix bulb bug

wjunjie01 committed:
Add list command to command summary

wjunjie01 committed:
Add inflow and outflow command to command summary

EdmundTangg committed:
Update UG and DG for add, find, history commands

EdmundTangg committed:
Merge pull request #91 from wjunjie01/ug-list

Update UG for list command

EdmundTangg committed:
solved merge conflict

Kobot7 committed:
Merge pull request #92 from EdmundTangg/master

Updated UG and DG

Kobot7 committed:
Rearrange features section

Added RepoSense author tags as rearranging sections will affect
authorship

cheeseong2001 committed:
Merge pull request #93 from Kobot7/user-guide

Rearrange features section

SuveenE commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,430 @@
 
-## Acknowledgements
+---
+layout: page
```
Are these two lines formatted correctly? The following is shown for me.
`layout: page title: Developer Guide —`

SuveenE commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,430 @@
+</div>
+
+### Architecture
+<img src="images/ArchitectureDiagram.png" alt=""/>
```
Really good architecture diagram. Very concise!

SuveenE commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,430 @@
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+--------------------------------------------------------------------------------------------------------------------
 
```
Would it be better if you add a product description at the start?

SuveenE commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,430 @@
+1. Prerequisites: List all products using `list` command. There should be at least multiple products in the list.
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+2. Test case: `history 1`<br>
```
Good usage of code blocks through out the DG!

SuveenE commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,430 @@
+### UI component
+The **API** of this component is specified in [`Ui.java`](https://github.com/AY2324S2-CS2113T-T09-3/tp/tree/master/src/main/java/seedu/stockpal/ui/Ui.java)
+
+![Structure of the UI Component](images/UiClassDiagram.png)
```
Seems like the image path is not given correctly.
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/75742713/daa9f970-40e2-41ff-8daf-7b07de810248)

alvinnnnnnnnnn commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,430 @@
+
+--------------------------------------------------------------------------------------------------------------------
+
+## **Appendix: Instructions for manual testing**
```
maybe you could add a list of command summary?

alvinnnnnnnnnn replied:
maybe you could add a list of all the commands?

alvinnnnnnnnnn commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,430 @@
+**API** : [`Command.java`](https://github.com/AY2324S2-CS2113T-T09-3/tp/blob/master/src/main/java/seedu/stockpal/commands/Command.java)
+
+
+![Structure of the Commands Component](images/CommandsClassDiagram.png)
```
It seems like the image path was not specified correctly
![Uploading Screenshot 2024-04-03 at 12.44.24.png…]()

alvinnnnnnnnnn commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,430 @@
+The edit product feature is facilitated by `EditCommand` which extends `Command`.
+
+The following sequence diagram summarizes what happens when a user inputs a valid `edit` command.
+<img src="images/EditCommandSequenceDiagram.png" alt="EditCommandSequenceDiagram.png"/>
```
UML diagram looks very comprehensive and provides a clear understanding of the components!

clarencepohh replied:
[DG review] UML sequence diagram looks good and proper labelling and notations are used. However the font seems a little small and might create issues for viewing quality.

alvinnnnnnnnnn commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,430 @@
+   otherwise `No match found.` will be printed out. 
+
 
-## Instructions for manual testing
```
lack of instructions for error handling

clarencepohh commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,430 @@
+### UI component
+The **API** of this component is specified in [`Ui.java`](https://github.com/AY2324S2-CS2113T-T09-3/tp/tree/master/src/main/java/seedu/stockpal/ui/Ui.java)
+
+![Structure of the UI Component](images/UiClassDiagram.png)
```
[DG review] Image does not seem to render on github page deployment.

clarencepohh replied:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/110753446/5b0c976d-5753-4e27-b7e9-1b6a84f553b0)

clarencepohh replied:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/110753446/4e2ed6dc-923c-4b62-8ef8-cdeef41696b2)

clarencepohh commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,430 @@
+**API** : [`Command.java`](https://github.com/AY2324S2-CS2113T-T09-3/tp/blob/master/src/main/java/seedu/stockpal/commands/Command.java)
+
+
+![Structure of the Commands Component](images/CommandsClassDiagram.png)
```
[DG review] Image does not seem to render on github page deployment.

clarencepohh replied:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/110753446/4268940e-70c4-4552-9348-f798986faa58)

clarencepohh commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,430 @@
+    - Cons: `productList#updateProduct()` will be more lengthy. May require further abstraction.
+
+### List feature
+<img src="images/ListCommandClass.png" alt=""/>
```
[DG review] UML notations do not seem to follow course requirements:
1. access modifiers (public, private, etc. should be -, +, ...)
2. Class name should not have the Capital 'C' in front.

Understand that this UML diagram is likely generated from Intellij's PUML plugin. Maybe you could look into PUML syntax to change/remove these icons:
1. use `skinparam classAttributeIconSize 0` at the top of your .puml file.
2. use `hide circle` at the bottom of your .puml file.

clarencepohh commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,430 @@
+
+### InflowCommand Feature
+
+**API** : [`InflowCommand.java`](https://github.com/AY2324S2-CS2113T-T09-3/tp/blob/master/src/main/java/seedu/stockpal/commands/InflowCommand.java)
```
[DG review] Nice use of in-line links for devs to refer to code base!

clarencepohh commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,430 @@
+<img src="images/HistoryCommandSequence.png" alt="HistoryCommandSequence.png"/>
+
+The following sequence diagram details how `HistoryCommand#execute()` functions.
+<img src="images/HistoryCommandExecuteSequenceDiagram.png" alt="HistoryCommandExecuteSequenceDiagram.png"/>
```
[DG review] Image does not appear on github page deployment.

clarencepohh replied:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/110753446/72e63ad6-bbd2-41d7-a305-1e73dd2d0810)

clarencepohh commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,430 @@
 
-## Acknowledgements
+---
+layout: page
+title: Developer Guide
```
[DG review] Formatting here looks incomplete.

clarencepohh replied:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/110753446/45696ed3-e479-4820-9862-1e832d3d1beb)

clarencepohh reviewed (commented):
Overall great documentation, just a few minor display issues on the page deployment!

Kobot7 committed:
Fix UG typo and formatting errors

Kobot7 committed:
Update icon formatting

Kobot7 committed:
Change page theme

Kobot7 committed:
Minor formatting error fix

Kobot7 committed:
Update UG for Running StockPal section

Kobot7 committed:
Fix size of images

Kobot7 committed:
Fix broken links in command summary

Kobot7 committed:
Update page theme

Kobot7 committed:
Updated README.md and Page Titles

Kobot7 committed:
Fix README.md title

Kobot7 committed:
Revert README.md title

Previous update caused Github pages to crash.

NgYaoDong committed:
Merge pull request #95 from Kobot7/user-guide

Update User guide

cheeseong2001 committed:
Add javadoc for Parser and DeleteCommand

cheeseong2001 committed:
Remove unused import

NgYaoDong committed:
Merge pull request #96 from cheeseong2001/branch-Parser

Add javadoc for Parser and DeleteCommand

EdmundTangg committed:
added HistoryCommandExecuteSequenceDiagram

NgYaoDong committed:
Merge pull request #97 from EdmundTangg/master

added HistoryCommandExecuteSequenceDiagram

NgYaoDong committed:
Update User Guide

Kobot7 committed:
Merge pull request #98 from NgYaoDong/NgYaoDong-UserGuide

Update User Guide

Ijaaz01 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,461 @@
+### UI component
+The **API** of this component is specified in [`Ui.java`](https://github.com/AY2324S2-CS2113T-T09-3/tp/tree/master/src/main/java/seedu/stockpal/ui/Ui.java)
+
+![Structure of the UI Component](images/UiClassDiagram.png)
```
Image is not showing up correctly probably because the name of the png was changed from 'uiClassDiagram' to just 'Ui' 
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/110718665/c08f4f1b-4fbd-4e76-a228-3c5ba7f0f17f)

jing-xiang replied:
rendering error

Ijaaz01 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,461 @@
+
+The *Sequence Diagram* below shows how the components interact with each other for the scenario where the user issues the command `delete 1`.
+
+<img src="images/ArchitectureSequenceDiagram.png" alt=""/>
```
Sequence diagram was clear and concise. Good job on not putting too much info on the diagram!

Ijaaz01 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,461 @@
+**API** : [`Data`](https://github.com/AY2324S2-CS2113T-T09-3/tp/blob/master/src/main/java/seedu/stockpal/data)
+
+The following is a class diagram of the `data` component.
+![Structure of the Data Component](images/DataClassDiagram.png)
```
Class diagram is very neat however I think its unnecessary to include all the methods. For example trivial methods like 'isEmpty()' or 'size()' can be omitted. Would help to clean up the class diagram a little

StargazerX0 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,461 @@
+
+**API** : [`Storage.java`](https://github.com/AY2324S2-CS2113T-T09-3/tp/blob/master/src/main/java/seedu/stockpal/storage/Storage.java)
+
+![Structure of the Storage Component](images/StorageClassDiagram.png)
```
ite seems the path of the images is not correct

Ijaaz01 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,461 @@
+* price: The price of the product.
+* description: The description of the product.
+
+**Methods**
+* `NewCommand`: Constructor for creating a new instance of the NewCommand class.
+* `execute`: Method to add the new product to the product list.
+* `createProduct`: Method to create a new product with a unique product ID.
+
+The following sequence diagram shows how an add operation works when the user inputs a valid `add` command:
+<img src="images/AddCommandSequence.png" alt="AddCommandSequence.png"/>
+
+The following sequence diagram details how `AddCommand#execute()` functions.
+<img src="images/AddCommandExecuteSequenceDiagram.png" alt="AddCommandExecuteSequenceDiagram.png"/>
```
Sequence diagram is very thorough. One small thing to fix would be the colour of the alt conditions as they blend with the dark red bars, making it a little difficult to read. could look into using a different colour

classskipper351 replied:
very precise sequence diagram however the size of  picture is a little small so I couldn't see very clearly, maybe shorten eh line between classes

StargazerX0 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,461 @@
+
+![Structure of the UI Component](images/UiClassDiagram.png)
+
+### Parser component
```
Nice explanation with link of code source, this help other developer to understand the code.

StargazerX0 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,461 @@
+The following sequence diagram details how `EditCommand#execute()` functions.
+<img src="images/EditCommandExecuteSequenceDiagram.png" alt="EditCommandExecuteSequenceDiagram.png"/>
+
+**Aspect: Validating parameters and handling errors**
```
Nice alternative suggestions, which give other developers the flexibility to handle error

Ijaaz01 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,461 @@
+* `sortListAccordingly`: Method to sort the list according to the products' PID, products' name or products' quantity.
+
+The following sequence diagram shows how a list operation works, by calling `list`.
+<img src="images/ListCommandSequence.png" alt=""/>
```
Sequence diagram is straight to the point good job. Appreciate the use of different colour arrows depending on which block they originate from. Only thing to keep in mind is to ensure that the arrows are still visible as they are quite thin!

StargazerX0 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,461 @@
+  * `CsvWriter` is responsible for handling the writing of data to the CSV data file.
+  * `CsvReader` is responsible for handling the reading of data from the CSV data file.
+
+### Common classes
```
The explanation of common classes is too brief, can give more detailed explanation them and provide some examples

Ijaaz01 reviewed (commented):
overall great work!

StargazerX0 reviewed (commented):
Well done, detailed and clear developer guide except that it has some display issues and some parts need more elaboration.

classskipper351 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,461 @@
+
+2. Test case: `new n/Drinking Cup q/20`<br>
+   Expected: The product will be added. Name of the product is `Drinking Cup`, 
+   Quantity of chocolate Milk stock is `20` units.
+
+3. Test case: `new n/Chocolate Milk q/100 p/2.00 d/Marigold HL Milk`<br>
+   Expected: The product will be added. Name of the product is `Chocolate Milk`,
+   Quantity of chocolate Milk stock is `100 ` units
+   Price of each unit is $`2.00`
+   Description of the Chocolate Milk product is `Marigold HL Milk`, which is the brand.
+
+### Editing Product Details
+1. Prerequisites: List all products using `list` command. There should be at least multiple products in the list.
+
+2. Test case: `edit 1 n/Updated name d/Updated description`<br> 
+   Expected: The name and description of the product with Product ID (PID) 1 
+   will be changed to `Updated name` and `Updated description` respectively.
+
+3. Test case: `edit 1 q/100 p/0.99`<br>
+   Expected: The quantity and price of the product with Product ID (PID) 1
+   will be changed to `100` and `0.99` respectively.
+
+### Deleting a product
+1. Prerequisites: List all products using `list` command. There should be a particular product with `pid` of 1 and no 
+product with `pid` of 2.
+
+2. Test case 1: `delete 1`. 
+   Expected: `product` with `pid` of 1 is deleted from the list. `"Product has been deleted"` is printed to the user.
+
+   Test case 2: `delete 2`.
+   Expected: `"Product with pid: 2 not found"`
+
+
+### Finding a keyword in the Product list 
+1. No prerequisites needed.
+
+2. Test case: `find Cor`<br>
+   Expected: A list of products will be printed out if there is a match, 
+   otherwise `No match found.` will be printed out. 
+
```
nice test cases to make commands more understandable

classskipper351 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,461 @@
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ... | I want to ...                                  | So that I can ...                                                                             |
+|---------|----------|------------------------------------------------|-----------------------------------------------------------------------------------------------|
+| v1.0    | new user | see usage instructions                         | refer to them when I forget how to use the application                                        |
+| v1.0    | new user | add details of products easily                 | track my products' stock                                                                      |
+| v1.0    | new user | entered data to be saved                       | track all the products that I have entered                                                    |
+| v1.0    | user     | be able to edit details of the products easily | easily change the name / price / quantity /description of the products, if there is an error  |
+| v1.0    | user     | have a small manual page                       | be aware of the commands that I can enter                                                     |
+| v2.0    | user     | view at past inflow/outflows of a product      | keep track of sales data of the product and know which products are of higher demand          |
+| v2.0    | user     | find products' name that contain the keyword   | find my product, or want to get the product's PID in a long list of products                  |
+| v2.0    | user     | find my products' past transactions            | keep track of the changes in my products' quantity                                            |
```
maybe specify the types of user

jing-xiang commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,461 @@
+2. Test case: `history 1`<br>
+   Expected:
+   A list of transactions will be printed out if there is a match,
+   otherwise `No match found.` will be printed out.
```
seems like a 404 error when i click subscribe at the bottom of the page

![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/61624545/7e43079e-b9ce-4cb7-8a00-a29cb95f8dbc)

jing-xiang commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,461 @@
+
+
+The following sequence diagram shows how a find operation works when the user inputs a valid `history` command:
+<img src="images/HistoryCommandSequence.png" alt="HistoryCommandSequence.png"/>
```
good color coding for this

![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/61624545/fa3bac92-a96c-4ee1-8cb4-9585013d0864)

jing-xiang commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,461 @@
+The sections below give more details of each component.
+
+### UI component
+The **API** of this component is specified in [`Ui.java`](https://github.com/AY2324S2-CS2113T-T09-3/tp/tree/master/src/main/java/seedu/stockpal/ui/Ui.java)
```
why is it called API?

jing-xiang reviewed (commented):
There are some image rendering errors. Not sure why some are named API. Diagrams are well done and detailed.

classskipper351 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,461 @@
+In the ProductList class, the `decreaseAmount` method will call a `updatedDecreaseQuantity`
+method in the quantity class.
+
+It is implemented this way to adhere to Single Responsibility Principle (SRP), such that the
+ProductList class will only handle the product involved in quantity increase, whereas the
+Quantity class will be responsible for updating the quantities.
```
nice illustration about why it is designed

wjunjie01 committed:
Fix sort feature in list bug

wjunjie01 committed:
Remove list class UML diagram

wjunjie01 committed:
Update user guide for extra inputs

wjunjie01 committed:
Add br to UserGuide

EdmundTangg committed:
Merge pull request #129 from wjunjie01/list

Fix sort feature in list, command usage for UG and class diagram

Kobot7 committed:
Update Edit Command description to indicate requirement of min 1 flag

Kobot7 committed:
Fix action bar of EditCommand sequence diagram

Kobot7 committed:
Update EditCommand sequence diagram image

Kobot7 committed:
Add "Notes about Command format" to UG Table of Contents

Kobot7 committed:
Remove page theme

Kobot7 committed:
Add details for UG Edit Command

EdmundTangg committed:
fixed bugs from pe dry run

EdmundTangg committed:
Merge pull request #161 from Kobot7/bug-fixes

Bug fixes for UG, DG and HelpCommand

EdmundTangg committed:
Edited DG to add in CommandSequenceDiagram

cheeseong2001 committed:
Fix parser regex rules to handle inputs with random whitespace placings

cheeseong2001 committed:
Refactor Exit command error message

cheeseong2001 committed:
Add more details on allow characters and exit command example in UG

cheeseong2001 committed:
Update parser regex for List and Find Command

cheeseong2001 committed:
Update Parser Class Diagram

EdmundTangg committed:
extra test cases for add command, added in PPP file

EdmundTangg committed:
Fixed merge conflict

EdmundTangg committed:
Merge branch 'EdmundTangg-bugfix'

EdmundTangg committed:
attempt to fix merge conflict

EdmundTangg committed:
attempt to fix merge conflict 2

EdmundTangg committed:
attempt to fix merge conflict 3

EdmundTangg committed:
fixed merge conflict

NgYaoDong committed:
Merge pull request #164 from EdmundTangg/bugfix

PE-D bug fixes, PPP

NgYaoDong committed:
Edit DG, draft PPP, and add fixes to bugs of PED.

NgYaoDong committed:
Remove redundant file

leongxingyu committed:
Merge pull request #168 from NgYaoDong/NgYaoDong-Docs

Edit DG, draft PPP, and add fixes to bugs of PED.

leongxingyu committed:
Update DG with Manual Testing

leongxingyu committed:
Edit um sequence diagram for inflow and outflow class

leongxingyu committed:
Update Inflow and Outflow command in DG

leongxingyu committed:
Update PPP

leongxingyu committed:
Edit formatting of headers

leongxingyu committed:
Edit typo in DG

leongxingyu committed:
Edit UML sequence diagram

leongxingyu committed:
Edit manual testing for Inflow command

leongxingyu committed:
Update Ui.png in readme

wjunjie01 committed:
Add list sort test file and code

leongxingyu committed:
Add in photo for AboutUs

cheeseong2001 committed:
Update parser logic to become iterative instead of one-time-match

cheeseong2001 committed:
Update parser and UG accordingly to new parser logic

NgYaoDong committed:
Merge pull request #174 from leongxingyu/Fix-PE-D-bugs

Add v2.1

NgYaoDong committed:
Add fix for bug when deleting product, but transaction still lingering.

cheeseong2001 committed:
Fix checkstyles

cheeseong2001 committed:
Merge branch 'master' into branch-BugFix

cheeseong2001 committed:
Merge pull request #175 from NgYaoDong/NgYaoDong-PED_Fix

Add fix for bug when deleting product, but transaction still lingering.

cheeseong2001 committed:
Merge branch 'master' into branch-BugFix

cheeseong2001 committed:
Update PPP

cheeseong2001 committed:
Fix more parser bugs

cheeseong2001 committed:
Update DG

wjunjie01 committed:
Improve sort type feature

wjunjie01 committed:
Update PPP and list UML diagram in DG

cheeseong2001 committed:
Make example for Delete in UG more specific

wjunjie01 committed:
Merge pull request #176 from cheeseong2001/branch-BugFix

Update Parser workflow to show more descriptive errors, update DG and PPP

wjunjie01 committed:
Merge branch 'master' into list

wjunjie01 committed:
Update PPP PR

wjunjie01 committed:
Update parser for list command

wjunjie01 committed:
Fix Gradle typo

wjunjie01 committed:
Upload picture

wjunjie01 committed:
Fix test typo

Kobot7 committed:
Add Project Portfolio Page

Kobot7 committed:
Update AboutUs.md with team information

Kobot7 committed:
Update PRODUCT_NAME constraints in User Guide

Kobot7 committed:
Created a more specific exception for duplicate product name error

Kobot7 committed:
Add EditCommand test case for duplicate product names

cheeseong2001 committed:
Merge branch 'master' into branch-BugFix

Kobot7 committed:
Update DG and fix duplicate product name logic

Kobot7 committed:
Add implementation details for HelpCommand to DG

Kobot7 committed:
Fix gradle check issue

- IntelliJ auto-converted imports to `import *` previously, causing gradle check to fail

Kobot7 committed:
Merge pull request #177 from wjunjie01/list

Update DG sequence diagram, PPP and add test coverage

cheeseong2001 committed:
Add more in-depth error messaging

cheeseong2001 committed:
Update Parser to throw more specific errors

cheeseong2001 committed:
Update DG and UG according to Parser behaviour

cheeseong2001 committed:
Merge pull request #178 from Kobot7/docs-update

Update DG, UG and PPP

cheeseong2001 committed:
Merge branch 'master' into branch-BugFix

# Conflicts:
#	docs/UserGuide.md
#	src/main/java/seedu/stockpal/common/Messages.java

cheeseong2001 committed:
Fix merge conflict

cheeseong2001 committed:
Fix link

cheeseong2001 committed:
Refactor methods

Kobot7 committed:
Merge pull request #180 from cheeseong2001/branch-BugFix

Update parser for more specific errors

Kobot7 committed:
Update command details for help page

Kobot7 committed:
Add details to DG for help feature

Kobot7 committed:
Fix action bar for DG sequence diagrams

Kobot7 committed:
Update userInput prompt and error message

Kobot7 committed:
Add line dividers to surround user input

EdmundTangg committed:
Merge pull request #181 from Kobot7/help-command-update

Update Help command & DG

EdmundTangg committed:
Fixed documentation bugs in DG

EdmundTangg committed:
Fixed fail checks

EdmundTangg committed:
edited typos

EdmundTangg committed:
deleted unnecessary files

NgYaoDong committed:
Merge pull request #182 from EdmundTangg/DG-bugFix

Fixed documentation bugs in DG

cheeseong2001 committed:
Update Parser to be case insensitive

cheeseong2001 committed:
Update UG to reflect case insensitivity update

NgYaoDong committed:
Merge pull request #183 from cheeseong2001/branch-BugFix

Update Parser to be case insensitive

NgYaoDong committed:
Fix detection of duplicate PID, names bug in storage file, and updated all documents.

NgYaoDong committed:
Fix anchoring of `commandFormat` in UG

Kobot7 committed:
Add more details to help's help page & update DG/UG accordingly

Kobot7 committed:
Merge pull request #184 from NgYaoDong/NgYaoDong-Docs

Fix detection of duplicate PID, names bug in storage file

Kobot7 committed:
Merge branch 'master' into help-description-update

Kobot7 committed:
Mark command keywords as code in UG Help Command description

Kobot7 committed:
Add back bullet points

Kobot7 committed:
Add missing closing list tag

Kobot7 committed:
Update UG

- Replaced command to run StockPal with a command that works on all Platforms
- Standardise description for quantity field
- Update inflow/outflow quantity description with constraint of >0
- Add missing full stops to some constraints.

NgYaoDong committed:
Merge pull request #185 from Kobot7/help-description-update

Add more details to help's help page & update DG/UG accordingly

Kobot7 committed:
Fix rendering of whitespace

Kobot7 committed:
Replace space with &nbsp;

NgYaoDong committed:
Merge pull request #186 from Kobot7/user-guide-fix

User guide fix

Kobot7 committed:
Fix list usage rendering in table

Kobot7 committed:
Update rendering of vertical bar

Kobot7 committed:
Fix missing rendering of whitespace

Kobot7 committed:
Remove indentation for inflow input example

EdmundTangg committed:
Merge pull request #187 from Kobot7/user-guide-fix

User guide fix

NgYaoDong committed:
Update UG and PPP

Kobot7 committed:
Merge pull request #188 from NgYaoDong/NgYaoDong-Docs

Update UG and PPP

EdmundTangg committed:
shortened PPP pages

EdmundTangg committed:
Edited grammar

Kobot7 committed:
Merge pull request #189 from EdmundTangg/master

Edited PPP

cheeseong2001 committed:
Update PPP

cheeseong2001 committed:
Minor bug fix to Inflow and Outflow command

cheeseong2001 committed:
Edit PPP

NgYaoDong committed:
Merge pull request #190 from cheeseong2001/branch-BugFix

Update PPP and minor bug fix to Parser

leongxingyu committed:
Edit PPP

leongxingyu committed:
Edit PPP to add mroe details

NgYaoDong committed:
Merge pull request #191 from leongxingyu/Edit-PPP

Edit ppp

wjunjie01 committed:
Edit PPP

NgYaoDong committed:
Merge pull request #192 from wjunjie01/master

Edit PPP

Kobot7 committed:
Change UG EditCommand description

Kobot7 committed:
Add instructions for manual testing for help page

NgYaoDong committed:
Merge pull request #193 from Kobot7/ug-dg-update

Docs update

Kobot7 committed:
Update help page for edit and find command

NgYaoDong committed:
Merge pull request #194 from Kobot7/update-help-page

Update help page for edit and find command
